tests/cases/fourslash/completionPreferredSuggestions1.ts

```diff
@@ -15,6 +15,6 @@
 
 verify.completions({ marker: "1", includes: ["a", "b", "c"] });
 verify.completions({ marker: "2", includes: ["0", "1", "2"], isNewIdentifierLocation: true });
-verify.completions({ marker: "3", includes: ["a", "b", "c"] });
+verify.completions({ marker: "3", exact: [] });
 verify.completions({ marker: "4", exact: [] });
-verify.completions({ marker: "5", includes: ["a", "b", "c"] });
+verify.completions({ marker: "5", exact: [] });

```

Comment:
Since `Record<never, never>` is just `{}` with this PR those are expected *here*. Those unions were reduced to just `string` and thus concrete literal strings can no longer be suggested.

Both of those changed locations obey the rule from https://github.com/microsoft/TypeScript/pull/49119 :
> For backwards compatibility, special exceptions to the T & {} type reduction rules existing for intersections written explicitly as string & {}, number & {}, and bigint & {} (as opposed to created through instantiation of a generic type T & {}).

Reply:
How is it that this `string & Record<never, never>` was being preserved before? Is it because, before this PR, `Record<never, never>` was some mapped type and we couldn't do subtype reduction on the intersection?

I'm curious to see if this breaks people, since this is in our tests, but you're right, officially we only support `string & {}` explicitly written as so.

Reply:
> Is it because, before this PR, Record<never, never> was some mapped type and we couldn't do subtype reduction on the intersection?

IIRC - yeah, something like that. The type was not reduced to `{}` so it stayed as a mapped type and that wasn't reduced. But with this change it's treated like `{}` so intersections containing it are way more easily reducible.